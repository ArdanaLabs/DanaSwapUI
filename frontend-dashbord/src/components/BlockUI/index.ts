export { default as BlockUI } from './BlockUI'