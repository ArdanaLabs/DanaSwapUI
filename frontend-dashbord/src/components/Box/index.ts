export { default as ProfileBox } from './ProfileBox'
export { default as OverViewBox } from './OverViewBox'
export { default as TokenBox } from './TokenBox'
export { default as StatBox } from './StatBox'