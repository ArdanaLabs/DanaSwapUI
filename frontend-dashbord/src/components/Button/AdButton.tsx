import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";

const StyledButton = withStyles({
  root: {
    margin: "10px",
    padding: "15px 30px",
    borderRadius: "50px",
    fontSize: "18px",
    fontWeight: 900,
    backgroundColor: "#FFFFFF",
    color: "#202F9A",
    whiteSpace: 'nowrap',
    width: "300px",
    "&:hover": {
      color: "#FFFFFF",
      backgroundColor: "#202F9A",
    },
  },
})(Button);

export default StyledButton;
