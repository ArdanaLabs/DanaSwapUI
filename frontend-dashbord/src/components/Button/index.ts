export { default as Button } from './Button'
export { default as Radio } from './Radio'
export { default as DropdownButton } from './DropdownButton'
export { default as AdButton } from './AdButton'
export { default as SwapButton } from './SwapButton'