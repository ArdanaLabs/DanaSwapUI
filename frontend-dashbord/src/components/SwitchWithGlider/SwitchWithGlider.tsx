import React from 'react';
import { Box } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(({ palette }) => ({
  container: {
    display: 'flex',
    border: palette.background.paper,
    width: '100%',
    height: '100%',
    justifyContent: 'space-between'
  },
  activeItem: {
    borderRadius: '100px 0 0 100px',
    background: '#FFFFFF',
    transition: 'background .3s ease-in',
  }

  // glider: {
  //   transition: 'all 0.4s ease-out',
  //   position: 'absolute',
  //   borderRadius: '10px',
  //   backgroundColor: palette.primary.dark
  // }
}));

export interface SwitchWithGliderProps {
  elements: any;
  defaultIndex: number;
  gliderHeight: number;
  gliderWidth: number | string;
  marginBetweenSwitches: number;
  verticalGlider?: boolean;
}

const SwitchWithGlider: React.FC<SwitchWithGliderProps> = ({
  elements,
  gliderHeight,
  gliderWidth,
  defaultIndex,
  marginBetweenSwitches,
  verticalGlider
}) => {
  const classes = useStyles();
  // const [gliderPosition, setGliderPosition] = React.useState<any>(0);

  // React.useEffect(() => {
  //   if (!verticalGlider) {
  //     const incrementalDistance =
  //       (typeof gliderWidth === 'number' ? gliderWidth : 0) +
  //       marginBetweenSwitches;
  //     const newPosition = defaultIndex * incrementalDistance;
  //     setGliderPosition(newPosition);
  //   } else {
  //     const incrementalDistance = gliderHeight + marginBetweenSwitches;
  //     const newPosition = defaultIndex * incrementalDistance;
  //     setGliderPosition(newPosition);
  //   }
  // }, [
  //   defaultIndex,
  //   gliderHeight,
  //   gliderWidth,
  //   marginBetweenSwitches,
  //   verticalGlider
  // ]);

  const wrappedElements = elements.map((item: React.FC, index: number) =>
    index !== defaultIndex ? (
      <Box key={index}>{item}</Box>
    ) : (
      <Box className={classes.activeItem} key={index}>
        {item}
      </Box>
    )
  );

  return (
    <Box width="100%" height="100%">
      {/* <Box
        className={classes.glider}
        width={gliderWidth}
        height={gliderHeight}
        style={
          !verticalGlider
            ? {
                transform: `translateX(${gliderPosition}px)`,
                opacity: gliderPosition < 0 ? 0 : 1
              }
            : {
                transform: `translateY(${gliderPosition}px)`,
                borderRadius: '12px',
                opacity: gliderPosition < 0 ? 0 : 1
              }
        }
      /> */}
      <Box
        className={classes.container}
        flexDirection={!verticalGlider ? 'row' : 'column'}
        alignItems={!verticalGlider ? 'center' : 'flex-start'}
      >
        {wrappedElements}
      </Box>
    </Box>
  );
};

export default SwitchWithGlider;
