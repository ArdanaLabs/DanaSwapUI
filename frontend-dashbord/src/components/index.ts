export { default as ThemeSwitch } from './ThemeSwitch';
export { default as SwitchWithGlider } from './SwitchWithGlider';
export { default as Slider } from './Slider';
export * from './Table';
export * from './BlockUI';