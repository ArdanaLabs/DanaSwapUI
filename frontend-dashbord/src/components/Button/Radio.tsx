import React from "react";
import {
  Box,
  useMediaQuery,
  Radio,
  RadioGroup,
  FormControlLabel,
  FormControl,
  FormLabel,
} from "@material-ui/core";
import { makeStyles, useTheme } from "@material-ui/core/styles";
import cx from "classnames";
import { useIsDarkMode } from "state/user/hooks";
import { Input } from "components/Input";

const useStyles = makeStyles(({ palette }) => ({
  title: {
    fontSize: "10px",
    color: palette.text.primary,
    fontWeight: "bold",
    marginBottom: "4px",
  },

  root: {
    "&:hover": {
      backgroundColor: "transparent",
    },
  },
  icon: {
    borderRadius: "50%",
    width: 18,
    height: 18,
    background: palette.type === "light" ? "#E9E9E9" : palette.common.white,
    transition: "background .3s ease-in",
    "input:hover ~ &": {
      // backgroundColor: "#ebf1f5",
    },
    "input:disabled ~ &": {
      boxShadow: "none",
      background: "rgba(206,217,224,.5)",
    },
  },
  checkedIcon: {
    background: palette.type === "light" ? palette.primary.main : palette.text.hint,
    // "&:before": {
    //   display: "block",
    //   width: 18,
    //   height: 18,
    //   backgroundImage: `url(${Check})`,
    //   backgroundRepeat: "no-repeat",
    //   backgroundPosition: "4px",
    //   content: '""',
    // },
    "input:hover ~ &": {
      // backgroundColor: "#206ba3",
    },
  },
  itemLabel: {
    fontSize: "10px",
    color: palette.text.secondary,
  },
  formControl: {
    "& .MuiRadio-root": {
      padding: "5px 9px",
    },
  },
  customInput: {
    fontSize: "7px",
    width: "calc(100% - 20px)",
    padding: "4px 2px 4px 9px",
    background: palette.type === 'light' ? '#E5E5E5' : "white",
  },
}));

export interface CustomRadioProps {
  option: any;
  value: any;
  customValue?: any;
}

const CustomRadio: React.FC<CustomRadioProps> = ({
  option,
  value,
  customValue = "",
}) => {
  const { breakpoints } = useTheme();
  const dark = useIsDarkMode();
  const mobile = useMediaQuery(breakpoints.down("xs"));
  const classes = useStyles({ dark, mobile });
  const [val, setVal] = React.useState(value);
  const [customVal, setCustomVal] = React.useState(customValue);

  const onCustomInputChange = (e: any) => {
    setCustomVal(e.target.value);
  };
  const StyledRadio = (props: any) => {
    return (
      <Radio
        className={classes.root}
        disableRipple
        color="default"
        checkedIcon={<span className={cx(classes.icon, classes.checkedIcon)} />}
        icon={<span className={classes.icon} />}
        {...props}
      />
    );
  };

  const handleChange = (event: any) => {
    // console.log(event.target);
    setVal(event.target.value);
  };

  return (
    <FormControl>
      <FormLabel className={cx(classes.title)}>{option.title}</FormLabel>
      <RadioGroup
        aria-label={option.title}
        name={option.title}
        value={val}
        onChange={handleChange}
      >
        {option.data.map((item: any, i: any) => (
          <Box key={i}>
            <FormControlLabel
              className={cx(classes.formControl)}
              value={item.value}
              control={<StyledRadio />}
              label={
                item.hasInput ? (
                  <Input
                    placeholder={"Custom Amount"}
                    value={customVal}
                    className={cx(classes.customInput)}
                    onChange={onCustomInputChange}
                    type={"number"}
                  />
                ) : (
                  <div className={cx(classes.itemLabel)}>{item.label}</div>
                )
              }
            />
          </Box>
        ))}
      </RadioGroup>
    </FormControl>
  );
};

export default CustomRadio;
