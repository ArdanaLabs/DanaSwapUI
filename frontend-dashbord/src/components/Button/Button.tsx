import { withStyles } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'

const StyledButton = withStyles({
    root: {
        padding: '4px 43px 7px 43px',
        borderRadius: '25px',
        backgroundColor: '#006DFF',
        '&:hover': {
          backgroundColor: '#1976d2'
        }
    },
  })(Button);


export default StyledButton
