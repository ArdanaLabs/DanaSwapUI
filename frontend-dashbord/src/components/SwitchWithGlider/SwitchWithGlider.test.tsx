import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import * as Enzyme from "enzyme";
import { shallow } from "enzyme";
import SwitchWithGlider from ".";
import { Button } from "components/Button";

Enzyme.configure({ adapter: new Adapter() });

let wrapper: Enzyme.ShallowWrapper;
const handleSwitch = jest.fn();

describe("Components / SwitchWithClider / SwitchWithClider", () => {
  beforeEach(() => {
    wrapper = shallow(
      <SwitchWithGlider
        elements={["VOLUME", "TVL", "LIQUIDITY"]}
        normalClass={"normalClass"}
        activeClass={"activeClass"}
        activeIndex={0}
        handleSwitch={handleSwitch}
      />
    );
  });

  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have three Button", () => {
    expect(wrapper.find(Button)).toHaveLength(3);
  });
  it("should have active status", () => {
    wrapper.find(Button).at(1).simulate("click");
    expect(handleSwitch).toHaveBeenCalledTimes(1);
    // expect(wrapper.find(Button).at(0).hasClass("activeClass")).toEqual(true);
  });
});
