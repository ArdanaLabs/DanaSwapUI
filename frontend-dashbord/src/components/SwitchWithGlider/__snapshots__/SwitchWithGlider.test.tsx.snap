// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / SwitchWithClider / SwitchWithClider renders 1`] = `ShallowWrapper {}`;
